### PR TITLE
Migrate task for old arf reports, instead of db:migrate

### PR DESCRIPTION
**Pull request by shlomizadok**
This PR removes migrations related to ARF reports, as in current migration we are losing the xml files.
In the proposed PR, the ARF migration is removed in favor of a rake task which takes the **old** arf reports, sends it to the proxy for re-processing and generate it back as ForemanOpenscap::ArfReport (the **New** one), while saving the XML files in the proxy. This PR is depending on https://github.com/theforeman/smart_proxy_openscap/pull/25 PR).

todo:
- [ ] Tests (Now sure how to add tests for this...)
- [x] Remove Migration code entirely (some remarked, some removed)

Usage would be:
`rake foreman_openscap:migrate[<proxy_id>]`
So if the proxy ID with openscap feature would be 1, I'd run:
`rake foreman_openscap:migrate[1]`

### CODE REVIEW

File: lib/foreman_openscap/data_migration.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      foreman_status_url = Setting[:foreman_url] + '/status'
+      response = JSON.parse(RestClient.get foreman_status_url)
+      return true if response["status"] == "ok"
+    rescue *::ProxyAPI::AvailableProxy::HTTP_ERRORS
+      return false
+    end
+
+    def fetch_xml_file(id)
+      query = ActiveRecord::Migration.execute("SELECT bzip_data FROM foreman_openscap_arf_report_raws WHERE arf_report_id=#{id} LIMIT 1;")
+      ActiveRecord::Base.connection.unescape_bytea(query.first['bzip_data'])
```
Comment (unorthodoxgeek) [inline lib/foreman_openscap/data_migration.rb]: by tea? o.O

File: app/lib/proxy_api/migration.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module ::ProxyAPI
+  class Migration < ::ProxyAPI::Resource
+    def initialize(args)
+      @url = args[:url] + '/compliance-importer'
+      super args
+      @connect_params[:headers].merge!(:content_type => 'text/xml', :content_encoding => 'x-bzip2', :multipart => true)
+
```
Comment (ares) [inline app/lib/proxy_api/migration.rb]: nitpick, extra empty line

File: lib/tasks/foreman_openscap_tasks.rake [hunk 3]
```diff
@@ -17,6 +17,15 @@ namespace :foreman_openscap do
     end
   end
 
+  task :migrate, [:proxy] => [:environment] do |task, args|
+    require 'foreman_openscap/data_migration'
+    puts args[:proxy]
```
Comment (ares) [inline lib/tasks/foreman_openscap_tasks.rake]: a better string should be printed here

File: db/migrate/20150929152345_move_arf_reports_to_reports_table.rb [hunk 4]
```diff
@@ -3,60 +3,15 @@ class MoveArfReportsToReportsTable < ActiveRecord::Migration
   # rubocop:disable Metrics/MethodLength
   # rubocop:disable Metrics/AbcSize
   def up
-    old_arf_reports = execute("SELECT * FROM foreman_openscap_arf_reports;")
-
-    #select only reports with existing host
-    old_arf_reports = old_arf_reports.select do |item|
-      asset = ForemanOpenscap::Asset.find item['asset_id']
-      !asset.host.nil? && asset.assetable_type = "Host::Base"
```
Comment (ares) [inline db/migrate/20150929152345_move_arf_reports_to_reports_table.rb]: oh, you've fixed another bug by this, I guess the one `=` was not intentional here :-)

File: lib/foreman_openscap/data_migration.rb [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+      end
+    end
+
+    private
+
+    def foreman_available?
+      foreman_status_url = Setting[:foreman_url] + '/status'
+      response = JSON.parse(RestClient.get foreman_status_url)
+      return true if response["status"] == "ok"
+    rescue *::ProxyAPI::AvailableProxy::HTTP_ERRORS
```
Comment (ares) [inline lib/foreman_openscap/data_migration.rb]: I think `*` should not be here

Comment (ares) [inline lib/foreman_openscap/data_migration.rb]: ah my bad, I didn't realize HTTP_ERRORS is an array

File: lib/tasks/foreman_openscap_tasks.rake (hunk 3 above)
Comment (shlomizadok) [inline lib/tasks/foreman_openscap_tasks.rake]: This should be removed :laughing:

**Comment by ohadlevy** [comment]
@ares anything standing out before merge? thanks!

**Comment by ares** [comment]
I thought Tests checkox should be checked first. @shlomizadok do you plan to add tests for ForemanOpenscap::DataMigration#migrate that I was talking about? It's OK without tests as well.

**Comment by ares** [comment]
no tests will be added atm, thanks @shlomizadok, merging!

**Pull request merged**